biomage/stage/stage.py

```diff
@@ -188,19 +186,20 @@ def get_branch_ref(chart, token, repo_to_ref=None, return_sha=False):
         ref = repo_to_ref[repo_name]
 
     if isinstance(ref, int):
-        target_branch = f"refs/pull/{ref}/head"
+        pr = repo.get_pull(ref)
+        target_branch = pr.head.ref
```

Comment:
try to set the branch name in UI, api, worker, pipeline and remove the code in here